tests/unit/test_configurator/test_configurator_traces_exporter.py

```diff
@@ -0,0 +1,188 @@
+# © 2023 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+import os
+import pytest
+
+from solarwinds_apm import configurator
+
+# otel fixtures
+from .fixtures.batch_span_processor import fixture_mock_bsprocessor
+from .fixtures.trace import get_trace_mocks
+
+# apm python fixtures
+from .fixtures.apm_config import (
+    fixture_mock_apmconfig_disabled,
+    fixture_mock_apmconfig_enabled,
+)
+from .fixtures.extension import (
+    fixture_mock_reporter,
+)
+from .fixtures.fwkv_manager import fixture_mock_fwkv_manager
+from .fixtures.meter_manager import fixture_mock_meter_manager
+from .fixtures.txn_name_manager import fixture_mock_txn_name_manager
+
+
+class TestConfiguratorTracesExporter:
+    def test_configure_traces_exporter_disabled(
+        self,
+        mocker,
+        mock_reporter,
+        mock_txn_name_manager,
+        mock_fwkv_manager,
+        mock_apmconfig_disabled,
+        mock_bsprocessor,
+    ):
+        # Mock Otel
+        mock_trace, mock_get_tracer_provider, mock_add_span_processor = get_trace_mocks(mocker)
+
+        test_configurator = configurator.SolarWindsConfigurator()
+        test_configurator._configure_traces_exporter(
+            mock_reporter,
+            mock_txn_name_manager,
+            mock_fwkv_manager,
+            mock_apmconfig_disabled,
+        )
+        mock_bsprocessor.assert_not_called()
+        mock_get_tracer_provider.assert_not_called()
+        mock_add_span_processor.assert_not_called()
+
+    def test_configure_traces_exporter_none(
+        self,
+        mocker,
+        mock_reporter,
+        mock_txn_name_manager,
+        mock_fwkv_manager,
+        mock_apmconfig_enabled,
+        mock_bsprocessor,
+    ):
+        # Save any EXPORTER env var for later
+        old_traces_exporter = os.environ.get("OTEL_TRACES_EXPORTER", None)
+        if old_traces_exporter:
+            del os.environ["OTEL_TRACES_EXPORTER"]
+
+        # Mock Otel
+        mock_trace, mock_get_tracer_provider, mock_add_span_processor = get_trace_mocks(mocker)
+
+        test_configurator = configurator.SolarWindsConfigurator()
+        test_configurator._configure_traces_exporter(
+            mock_reporter,
+            mock_txn_name_manager,
+            mock_fwkv_manager,
+            mock_apmconfig_enabled,
+        )
+        mock_bsprocessor.assert_not_called()
+        mock_get_tracer_provider.assert_not_called()
+        mock_add_span_processor.assert_not_called()
```

Comment:
For this `traces_exporter_none` test, does it mean if `OTEL_TRACES_EXPORTER` env var is not set, our custom distro does not configure the solarwinds_exporter either?

Reply:
Ooh yes good question.

For `_configure_traces_exporter` in complete isolation, yes it could be that `OTEL_TRACES_EXPORTER` is None so this test is for that scenario.

For `_configure_traces_exporter` running at distro startup, that scenario shouldn't happen. [SolarWindsDistro._configure](https://github.com/solarwinds/apm-python/blob/b97fff717a84e9bd9be5efd22f882f35f5ccac05/solarwinds_apm/distro.py#L45-L53) is called first to set env var [defaults](https://github.com/solarwinds/apm-python/blob/b97fff717a84e9bd9be5efd22f882f35f5ccac05/solarwinds_apm/apm_constants.py#L19-L23). Outside lambda the default is the SW exporter. Then [SolarWindsConfigurator._configure](https://github.com/solarwinds/apm-python/blob/b97fff717a84e9bd9be5efd22f882f35f5ccac05/solarwinds_apm/configurator.py#L80) is called to init the exporters etc. If `OTEL_TRACES_EXPORTER` wasn't set by user, then default from distro is used instead.

Reply:
Ah gotcha, thanks for the pointers :)